### ISSUE TITLE
Process gets killed during UMAP

### ISSUE THREAD

**Issue by jiannan-xu** [MaartenGr/BERTopic#1322]
Hello Maarten,

I use BERTopic for topic modeling with a 259, 4364 sentences corpus. This corpus is around 6GB in CSV format. 

I am running the code on a server that is equipped with Quadro RTX 5000 and 16GB RAM: 

```
    from bertopic import BERTopic
    from sklearn.feature_extraction.text import CountVectorizer
    from umap import UMAP
    from hdbscan import HDBSCAN
    from sentence_transformers import SentenceTransformer
    from bertopic.vectorizers import ClassTfidfTransformer
    sentence_model = SentenceTransformer("all-MiniLM-L6-v2",device='cuda')
    umap_model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0,
                      metric='cosine',low_memory=True,verbose=True,random_state=2022)
    hdbscan_model = HDBSCAN(min_samples=10, gen_min_span_tree=True, prediction_data=True)
    vectorizer_model = CountVectorizer(ngram_range=(1, 2), stop_words="english", min_df=5)
    ctfidf_model = ClassTfidfTransformer()
    #representation_model = KeyBERTInspired()
    topic_model = BERTopic(
      embedding_model=sentence_model,        
      umap_model=umap_model,                    
      hdbscan_model=hdbscan_model,           
      vectorizer_model=vectorizer_model,   
      ctfidf_model=ctfidf_model,             
      calculate_probabilities=False,
      verbose=True,
      low_memory=True,
      nr_topics=38,
      seed_topic_list=seed_topic_list
    )
  
    message = df[(df['before_hire']=='After Hired')]['message'].astype('str').to_list()
    topics_all, prob_all = topic_model.fit_transform(message)
```

I got following messages:

2.0.1+cu117
There are 1 GPU(s) available.
We will use the GPU: Quadro RTX 5000
Batches: 100%|███████████████████████████████████████████████████████████████████████████████| 81074/81074 [21:19<00:00, 63.37it/s]
2023-06-07 10:38:59,191 - BERTopic - Transformed documents to Embeddings
Batches: 100%|███████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 36.60it/s]
UMAP(angular_rp_forest=True, metric='cosine', min_dist=0.0, n_components=5, random_state=2022, verbose=True)
Wed Jun  7 10:39:12 2023 Construct fuzzy simplicial set
Wed Jun  7 10:39:15 2023 Finding Nearest Neighbors
Wed Jun  7 10:39:16 2023 Building RP forest with 64 trees
Killed

**Comment by MaartenGr** [comment]
It might just be that you did not have enough RAM available for using UMAP on such a large dataset. Instead, it might be worthwhile to use [cuML's UMAP](https://maartengr.github.io/BERTopic/getting_started/dim_reduction/dim_reduction.html#cuml-umap) instead. I believe it reduces the necessary RAM quite a bit and also makes use of the GPU. Another thing you could do is fit on a subset of the data and perform inference on the rest.

**Comment by MaartenGr** [comment]
Sure, you could do it as follows:

```python
from umap import UMAP
model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine')
reduced_embeddings = model.fit(embeddings)
other_reduced_embeddings = model.transform(other_embeddings)
```

Then you need to combine the embeddings and pass them to BERTopic. Make sure to then [skip over dimensionality reduction](https://maartengr.github.io/BERTopic/getting_started/dim_reduction/dim_reduction.html#skip-dimensionality-reduction). 

There is also a guide [here](https://colab.research.google.com/drive/1W7aEdDPxC29jP99GGZphUlqjMFFVKtBC?usp=sharing) for training BERTopic on large data.
